### PR TITLE
Yosys's simplify inside plugin

### PR DESCRIPTION
**Pull request by mandrys**
In this PR, _simplify()_ is copied from yosys to systemverilog plugin code, so we can use its modified version.
For example, we need it to support simplification of struct that has a range as a child.

Latest test yosys-systemverilog CI run: https://github.com/antmicro/yosys-systemverilog/actions/runs/4607334079.
Previous one (before the rebase): https://github.com/antmicro/yosys-systemverilog/actions/runs/4595939382.

### CODE REVIEW

**Review by kamilrakoczy — APPROVED** [review]

**Comment by mandrys** [comment]
I removed the part of the code that adds simplify support for structs and unions that have a range - this should be added in another PR.

**Review by mglb — CHANGES_REQUESTED** [review]
Please add a short note about `simplify.{cc,h}` to [README][readme].

Since we're going to extend this file, we'll need to move it from the third_party directory. This has low priority, so let's ignore it for now and handle this as a part of some file organization cleanup someday.
However, you can mention in the [README][readme] that we're going to extend this file instead of `simplify()` in UhdmAst and that it will be moved somewhere else in the future.

[readme]: https://github.com/chipsalliance/yosys-f4pga-plugins/blob/0f28c428eaa48b9b3ab1440b5e17960a70663e3b/systemverilog-plugin/third_party/yosys/README

File: systemverilog-plugin/third_party/yosys/simplify.h [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#include "frontends/ast/ast.h"
+#include "const2ast.h"
```
[the remark this answers is not in the document]
Comment (mandrys) [reply systemverilog-plugin/third_party/yosys/simplify.h]: Removed

File: systemverilog-plugin/UhdmAst.cc [hunk 2]
```diff
@@ -1058,7 +1059,7 @@ static void simplify_format_string(AST::AstNode *current_node)
     current_node->children[0] = AST::AstNode::mkconst_str(preformatted_string);
 }
 
-static void simplify(AST::AstNode *current_node, AST::AstNode *parent_node)
+static void custom_simplify(AST::AstNode *current_node, AST::AstNode *parent_node)
```
[the remark this answers is not in the document]
Comment (mandrys) [reply systemverilog-plugin/UhdmAst.cc]: Made changes as you requested and added a short description in the README.

**Review by mglb — APPROVED** [review]

**Pull request merged**